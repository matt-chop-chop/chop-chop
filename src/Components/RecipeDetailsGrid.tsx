import { fromatCookingTime } from "@/utils";
import { TimeIcon } from "@chakra-ui/icons";
import { Flex, Grid, Text } from "@chakra-ui/react";

type RecipeDetailsGridProps = {
  area: string;
  category: string;
  ingredientCount: number;
  time: number;
};

const RecipeDetailsGrid = ({
  area,
  category,
  ingredientCount,
  time,
}: RecipeDetailsGridProps) => {
  return (
    <Grid
      columnGap={[8, 8, 6]}
      gridAutoRows="1fr"
      gridTemplateColumns={[
        "repeat(2, 1fr)",
        "repeat(2, 1fr)",
        "repeat(4, 1fr)",
      ]}
      mt={8}
      rowGap={8}
    >
      <Flex flexDirection="column">
        <Flex alignItems="center">
          <Text variant="body-bold">Cooking Time</Text>
          <TimeIcon boxSize={[3, 4, 4]} ml={[1, 2, 2]} />
        </Flex>
        <Text mt={1} variant="body">
          {fromatCookingTime(time)}
        </Text>
      </Flex>
      <Flex flexDirection="column">
        <Text variant="body-bold">Ingredient Count</Text>
        <Text mt={1} variant="body">
          {ingredientCount}
        </Text>
      </Flex>
      <Flex flexDirection="column">
        <Text variant="body-bold">Meal Type</Text>
        <Text mt={1} variant="body">
          {category}
        </Text>
      </Flex>
      <Flex flexDirection="column">
        <Text variant="body-bold">Place of Origin</Text>
        <Text mt={1} variant="body">
          {area}
        </Text>
      </Flex>
    </Grid>
  );
};

export default RecipeDetailsGrid;
