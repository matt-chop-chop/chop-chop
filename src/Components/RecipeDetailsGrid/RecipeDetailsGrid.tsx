import { fromatCookingTime } from "@/utils";
import { TimeIcon } from "@chakra-ui/icons";
import { Flex, Grid, Text } from "@chakra-ui/react";
import React from "react";

export type RecipeDetailsGridProps = {
  area: string;
  category: string;
  ingredientCount: number;
  time: number;
  gridTemplateColumns?: string[];
};

const RecipeDetailsGrid = ({
  area,
  category,
  ingredientCount,
  time,
  gridTemplateColumns = ["repeat(2, 1fr)", "repeat(2, 1fr)", "repeat(4, 1fr)"],
}: RecipeDetailsGridProps) => {
  return (
    <Grid
      columnGap={[8, 8, 6]}
      gridAutoRows="1fr"
      gridTemplateColumns={gridTemplateColumns}
      mt={8}
      rowGap={8}
    >
      <Flex flexDirection="column">
        <Flex alignItems="center">
          <Text variant="body-bold">Cooking Time</Text>
          <TimeIcon boxSize={[3, 4, 4]} ml={[1, 2, 2]} />
        </Flex>
        <Text mt={1} variant="body">
          {fromatCookingTime(time)}
        </Text>
      </Flex>
      <Flex flexDirection="column">
        <Text variant="body-bold">Ingredient Count</Text>
        <Text mt={1} variant="body">
          {ingredientCount}
        </Text>
      </Flex>
      <Flex flexDirection="column">
        <Text variant="body-bold">Meal Type</Text>
        <Text mt={1} variant="body">
          {category}
        </Text>
      </Flex>
      <Flex flexDirection="column">
        <Text variant="body-bold">Place of Origin</Text>
        <Text mt={1} variant="body">
          {area}
        </Text>
      </Flex>
    </Grid>
  );
};

export default RecipeDetailsGrid;
