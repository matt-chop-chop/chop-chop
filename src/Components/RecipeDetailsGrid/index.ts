export { default as RecipeDetailsGrid } from "./RecipeDetailsGrid";
