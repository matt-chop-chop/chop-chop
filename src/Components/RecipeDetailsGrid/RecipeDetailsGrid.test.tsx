import { shallow } from "enzyme";
import React from "react";
import RecipeDetailsGrid, { RecipeDetailsGridProps } from "./RecipeDetailsGrid";

const props: RecipeDetailsGridProps = {
  area: "Test Area",
  category: "Test Category",
  ingredientCount: 1,
  time: 10,
};

describe("RecipeDetailsGrid", () => {
  it("is exported from RecipeDetailsGrid", () => {
    expect(RecipeDetailsGrid).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(
      <RecipeDetailsGrid
        {...props}
        gridTemplateColumns={[
          "repeat(1, 1fr)",
          "repeat(2, 1fr)",
          "repeat(3, 1fr)",
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders with default props ", () => {
    const wrapper = shallow(<RecipeDetailsGrid {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
