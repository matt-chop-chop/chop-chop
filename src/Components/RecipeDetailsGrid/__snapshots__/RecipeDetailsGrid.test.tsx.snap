// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetailsGrid renders as expected  1`] = `
<Grid
  columnGap={
    [
      8,
      8,
      6,
    ]
  }
  gridAutoRows="1fr"
  gridTemplateColumns={
    [
      "repeat(1, 1fr)",
      "repeat(2, 1fr)",
      "repeat(3, 1fr)",
    ]
  }
  mt={8}
  rowGap={8}
>
  <Flex
    flexDirection="column"
  >
    <Flex
      alignItems="center"
    >
      <Text
        variant="body-bold"
      >
        Cooking Time
      </Text>
      <TimeIcon
        boxSize={
          [
            3,
            4,
            4,
          ]
        }
        ml={
          [
            1,
            2,
            2,
          ]
        }
      />
    </Flex>
    <Text
      mt={1}
      variant="body"
    >
      10 minutes
    </Text>
  </Flex>
  <Flex
    flexDirection="column"
  >
    <Text
      variant="body-bold"
    >
      Ingredient Count
    </Text>
    <Text
      mt={1}
      variant="body"
    >
      1
    </Text>
  </Flex>
  <Flex
    flexDirection="column"
  >
    <Text
      variant="body-bold"
    >
      Meal Type
    </Text>
    <Text
      mt={1}
      variant="body"
    >
      Test Category
    </Text>
  </Flex>
  <Flex
    flexDirection="column"
  >
    <Text
      variant="body-bold"
    >
      Place of Origin
    </Text>
    <Text
      mt={1}
      variant="body"
    >
      Test Area
    </Text>
  </Flex>
</Grid>
`;

exports[`RecipeDetailsGrid renders with default props  1`] = `
<Grid
  columnGap={
    [
      8,
      8,
      6,
    ]
  }
  gridAutoRows="1fr"
  gridTemplateColumns={
    [
      "repeat(2, 1fr)",
      "repeat(2, 1fr)",
      "repeat(4, 1fr)",
    ]
  }
  mt={8}
  rowGap={8}
>
  <Flex
    flexDirection="column"
  >
    <Flex
      alignItems="center"
    >
      <Text
        variant="body-bold"
      >
        Cooking Time
      </Text>
      <TimeIcon
        boxSize={
          [
            3,
            4,
            4,
          ]
        }
        ml={
          [
            1,
            2,
            2,
          ]
        }
      />
    </Flex>
    <Text
      mt={1}
      variant="body"
    >
      10 minutes
    </Text>
  </Flex>
  <Flex
    flexDirection="column"
  >
    <Text
      variant="body-bold"
    >
      Ingredient Count
    </Text>
    <Text
      mt={1}
      variant="body"
    >
      1
    </Text>
  </Flex>
  <Flex
    flexDirection="column"
  >
    <Text
      variant="body-bold"
    >
      Meal Type
    </Text>
    <Text
      mt={1}
      variant="body"
    >
      Test Category
    </Text>
  </Flex>
  <Flex
    flexDirection="column"
  >
    <Text
      variant="body-bold"
    >
      Place of Origin
    </Text>
    <Text
      mt={1}
      variant="body"
    >
      Test Area
    </Text>
  </Flex>
</Grid>
`;
